### PR TITLE
Add multi user cookie storage and unauthenticated transport session

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* This PR adds support to store different cookies in the keychain. Until now we only used one account name `"User"`, now we use a `NSUUID` passed to the initialiser  of `ZMPersistentCookieStorage`.
Migration of the old data is done by `ZMPersistentCookieStorageMigrator` which should be used to create the new storage as it will migrate old data if needed. 

* This PR also adds a new class `UnauthenticatedTransportSession` which should be used until a user is logged in and a cookie is present.

### CODE REVIEW

**Comment by daehn** [comment]
I will have to adopt the changes made in https://github.com/wireapp/wire-ios-transport/pull/43, which injects the cookie storage into the transport session. This will also make the migration handling a bit easier.

File: Source/Authentication/ZMPersistentCookieStorage.m [hunk 1]
```diff
@@ -107,7 +109,7 @@ - (NSData *)authenticationCookieData;
     if ([self findItemWithPassword:&result]) {
         return result;
     }
-    return nil;
+    return nil;
```
Comment (daehn) [inline Source/Authentication/ZMPersistentCookieStorage.m]: 😱

**Comment by daehn** [comment]
@vytis @typfel can we merge this one?

**Comment by daehn** [comment]
Or maybe let's wait, as it will break other parts

**Comment by vytis** [comment]
I think so!

**Comment by daehn** [comment]
Waiting until the first working login implementation using an in memory persistent store is merged.

**Comment by daehn** [comment]
@vytis @typfel I added the first parts of the `UnauthenticatedTransportSession`, please have a look@

**Comment by typfel** [comment]
shouldn't the filename be ` UnauthenticatedTransportSession.swift`?

File: Source/TransportSession/UnauthenticatedSession.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        self.baseURL = baseURL
+        super.init()
+    }
+
+    public func enqueueRequest(withGenerator generator: ZMTransportRequestGenerator) -> ZMTransportEnqueueResult {
+
+        func decrement(notify: Bool) {
+            let newCount = withUnsafeMutablePointer(to: &numberOfRunningRequests, OSAtomicDecrement32)
+            guard newCount < maximumNumberOfRequests, notify else { return }
+            ZMTransportSession.notifyNewRequestsAvailable(self)
```
Comment (typfel) [inline Source/TransportSession/UnauthenticatedSession.swift]: I can't find any one subscribing to these notifications.

Comment (daehn) [inline Source/TransportSession/UnauthenticatedSession.swift]: The operation loop using this transport session which is responsible for enqueuing requests  should subscribe to them, but that's in `wire-ios-sync-engine`.

File: Source/TransportSession/UnauthenticatedSession.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    private func parseCookie(from response: HTTPURLResponse) {
+        guard let data = response.extractCookieData() else { return }
+        delegate?.session(self, cookieDataBecomeAvailable: data)
+    }
+
+}
+
+
+extension UnauthenticatedTransportSession: URLSessionDelegate {
```
Comment (daehn) [inline Source/TransportSession/UnauthenticatedSession.swift]: Forgot to add the `UnauthenticatedTransportSession` as delegate of the `URLSession`

**Comment by daehn** [comment]
@typfel @vytis I cleaned up the code and added tests (also fixed the filename), please have a look!

**Comment by daehn** [comment]
I had to change the base to `develop` as the previous base got merged. Will rebase this one.

**Pull request merged**